src/org/labkey/test/SuiteBuilder.java

```diff
@@ -176,14 +179,54 @@ public TestSet getTestSet(String suiteName)
             optional = true;
             suiteName = suiteName.substring(1);
         }
+        // Support syntax for splitting up test suites
+        // e.g. "Daily[1/3]" to select the first third of tests in the Daily suite
+        int subset = 1;
+        int subsetCount = 1;
+        Pattern pattern = Pattern.compile("(.+)\\[(\\d+)/(\\d+)]");
+        Matcher matcher = pattern.matcher(suiteName);
+        if (matcher.matches())
+        {
+            subset = Integer.parseInt(matcher.group(2));
+            subsetCount = Integer.parseInt(matcher.group(3));
+            if (subset < 1 || subsetCount < 1 || subset > subsetCount)
+            {
+                throw new IllegalArgumentException("Invalid subsuite specification: " + suiteName);
+            }
+
+            suiteName = matcher.group(1);
+        }
         Set<Class> tests = _suites.getOrDefault(suiteName, optional ? Collections.emptySet() : null);
 
+        tests = extractSubset(tests, subset, subsetCount);
+
         if (tests == null)
-            throw new IllegalArgumentException(suiteName + " is not a valid test suite");
+            return null;
 
         return new TestSet(tests, suiteName);
     }
 
+    private Set<Class> extractSubset(Set<Class> tests, int subset, int subsetCount)
+    {
+        if (tests == null || tests.isEmpty() || subsetCount == 1)
+        {
+            return tests;
+        }
+
+        List<Class> sorted = new ArrayList<>(tests);
+        sorted.sort(Comparator.comparing(Class::getName));
+
+        int size = sorted.size();
+        int index = subset - 1;
+        int subsetSize = size / subsetCount;
+        int remainder = size % subsetCount;
+        int isLargeSubset = index < remainder ? 1 : 0;
+        int fromIndex = subsetSize * index + Math.min(index, remainder);
+        int toIndex = Math.min(fromIndex + subsetSize + isLargeSubset, size);
```

Comment:
When I do this math with a size of 100 and subsetCount = 3, the first batch is 0 to 34, the second is 34 to 67 and the third is 67 to 100.
In this particular case two test cases would be run twice (if my math is right). 
I don't think this is a big deal just something we should be aware of.

Otherwise I think this is fine.

Reply:
Your math is correct but `List.subList` doesn't include the element at `toIndex`.